src/prefect/orion/orchestration/core_policy.py

```diff
@@ -308,6 +311,14 @@ async def before_transition(
                 # Reset the run count so that the task run retries still work correctly
                 run.run_count = 0
 
+        # Reset pause metadata on retry
+        api_version = context.parameters.get("api-version", None)
+        if api_version is None or api_version >= Version("0.8.4"):
```

Comment:
Can you add a comment about why we are doing different behavior based on version here?